tests/modules/moduleconfigs.c

```diff
@@ -0,0 +1,139 @@
+#include "redismodule.h"
+#include <strings.h>
+int mutable_bool_val;
+int immutable_bool_val;
+long long longval;
+long long memval;
+RedisModuleString *strval = NULL;
+int enumval;
+
+/* Series of get and set callbacks for each type of config, these rely on the privdata ptr
+ * to point to the config, and they register the configs as such. Note that one could also just
+ * use names if they wanted, and store anything in privdata. */
+int getBoolConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    return (*(int *)privdata);
+}
+
+int setBoolConfigCommand(const char *name, int new, void *privdata, RedisModuleString **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    *(int *)privdata = new;
+    return REDISMODULE_OK;
+}
+
+long long getNumericConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    return (*(long long *) privdata);
+}
+
+int setNumericConfigCommand(const char *name, long long new, void *privdata, RedisModuleString **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    *(long long *)privdata = new;
+    return REDISMODULE_OK;
+}
+
+RedisModuleString *getStringConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(privdata);
+    return strval;
+}
+int setStringConfigCommand(const char *name, RedisModuleString *new, void *privdata, RedisModuleString **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    REDISMODULE_NOT_USED(privdata);
+    size_t len;
+    if (!strcasecmp(RedisModule_StringPtrLen(new, &len), "rejectisfreed")) {
+        *err = RedisModule_CreateString(NULL, "Cannot set string to 'rejectisfreed'", 36);
+        return REDISMODULE_ERR;
+    }
+    RedisModule_Free(strval);
```

Comment:
This line should be wrong, it should be `RedisModule_FreeString`, but it's not the cause of the test failure.

Reply:
it's interesting that valgrind doesn't report that leak (leaking the actual string).
i suppose the tcl test code doesn't set the string twice (replacing a previous one).

it's also odd that the alpine code crash on some corruption, but valgrind doesn't report anything.

Reply:
actually i see we do have a test that does it.
so i don't know why valgrind doesn't report the leak.
```
        r config set moduleconfigs.string wafflewednesdays
        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string wafflewednesdays"
        r config set moduleconfigs.string \x73\x75\x70\x65\x72\x20\x00\x73\x65\x63\x72\x65\x74\x20\x70\x61\x73\x73\x77\x6f\x72\x64
        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string {super \0secret password}"
```

Reply:
Yoav helped me figure this out. 
It must be an embedded string, so being short the robj pointer holds the sds in the same allocattion.

Reply:
I did test and confirm this was the case, I specifically asked for the case to make sure we weren't mismanaging the allocation, forgetting about embedded strings.